opamp-client/src/main/java/io/opentelemetry/opamp/client/internal/OpampClient.java

```diff
@@ -72,7 +77,7 @@ interface Callbacks {
      * @param client The relevant {@link OpampClient} instance.
      * @param throwable The exception.
      */
-    void onConnectFailed(OpampClient client, Throwable throwable);
+    void onConnectFailed(OpampClient client, @Nullable Throwable throwable);
```

Comment:
why do the callbacks pass in OpampClient? it seems that is available if needed already when registering the callback?